### PR TITLE
Add lib for load balancer and network interface

### PR DESCRIPTION
**Pull request by mikaelkrief**
### Description

Add librairies for:
- load balancer(s)
- network interface(s)

with lib, terraform, tests, docs and readme update with doc links

### CODE REVIEW

File: terraform/azure.tf [hunk 1]
```diff
@@ -634,3 +628,71 @@ resource "azurerm_sql_database" "sql-database" {
...
+  end_ip_address      = "${var.end_ip_address}"
+}
+
+resource "random_string" "lb-random" {
+  length  = 10
+  special = false
+  upper   = false
+}
+module "azurerm_lb" {
+  source              = "modules/LoadBalancer"
```
Comment (r-fennell) [inline terraform/azure.tf]: +1 for going with modules for this. I'd like to maintain `camel_case` for the modules though. Please rename to `load_balancer`

**Comment by josh-barker** [comment]
Hey @mikaelkrief, I'd love to use these resources, but the conflicting terraform files will stop the merge.

Could you take a look at resolving the conflicts?

**Comment by mikaelkrief** [comment]
@josh-barker , I fixed the merge issue

File: docs/resources/azurerm_load_balancer.md.erb [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.3.0 of the inspec-azure resource pack.
```
Comment (r-fennell) [inline docs/resources/azurerm_load_balancer.md.erb]: 1.4.0

File: docs/resources/azurerm_mysql_database.md.erb [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.3.0 of the inspec-azure resource pack.
```
Comment (r-fennell) [inline docs/resources/azurerm_mysql_database.md.erb]: 1.4.0

File: docs/resources/azurerm_mysql_databases.md.erb [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.3.0 of the inspec-azure resource pack.
```
Comment (r-fennell) [inline docs/resources/azurerm_mysql_databases.md.erb]: 1.4.0

File: docs/resources/azurerm_mysql_server.md.erb [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.3.0 of the inspec-azure resource pack.
```
Comment (r-fennell) [inline docs/resources/azurerm_mysql_server.md.erb]: 1.4.0

File: docs/resources/azurerm_mysql_servers.md.erb [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.3.0 of the inspec-azure resource pack.
```
Comment (r-fennell) [inline docs/resources/azurerm_mysql_servers.md.erb]: 1.4.0

File: docs/resources/azurerm_network_interface.md.erb [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.3.0 of the inspec-azure resource pack.
```
Comment (r-fennell) [inline docs/resources/azurerm_network_interface.md.erb]: 1.4.0

File: docs/resources/azurerm_network_interfaces.md.erb [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.3.0 of the inspec-azure resource pack.
```
Comment (r-fennell) [inline docs/resources/azurerm_network_interfaces.md.erb]: 1.4.0

File: docs/resources/azurerm_load_balancers.md.erb [hunk 9]
```diff
@@ -0,0 +1,143 @@
...
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.2.0 of the inspec-azure resource pack.
```
Comment (r-fennell) [inline docs/resources/azurerm_load_balancers.md.erb]: 1.4.0

File: docs/resources/azurerm_mysql_database.md.erb [hunk 10]
```diff
@@ -0,0 +1,124 @@
+---
+title: About the myazurerm_sql_database Resource
```
Comment (r-fennell) [inline docs/resources/azurerm_mysql_database.md.erb]: `title: About the azurerm_sql_database Resource`

File: docs/resources/azurerm_load_balancers.md.erb [hunk 11]
```diff
@@ -0,0 +1,143 @@
...
+    end
+<br />
+
+## Filter Criteria
+
+* `names`
+
+### names
+
+Filters the results to include only those servers which match the given name. This is a string value.
```
Comment (r-fennell) [inline docs/resources/azurerm_load_balancers.md.erb]: >...include only those *interfaces* which reside ...

File: terraform/modules/load_balancer/variables.tf [hunk 12]
```diff
@@ -0,0 +1,76 @@
+variable "location" {
+  description = "(Required) The location/region where the core network will be created. The full list of Azure regions can be found at https://azure.microsoft.com/regions"
+}
+
+variable "resource_group_name" {
+  description = "(Required) The name of the resource group where the load balancer resources will be placed."
+  default     = "azure_lb-rg"
```
Comment (r-fennell) [inline terraform/modules/load_balancer/variables.tf]: `inspec-azure_lb-rg` please, to make it more recognisable

File: terraform/modules/load_balancer/variables.tf [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+variable "lb_name" {
+  description = "name of the lb"
+}
+
+variable "use_loadbalancer" {
+  description = "Indique si on utilise un load balancer"
```
Comment (r-fennell) [inline terraform/modules/load_balancer/variables.tf]: en anglais s'il vous plait :)
>"Indicates if we are using a load balancer"

File: terraform/azure.tf [hunk 14]
```diff
@@ -635,6 +635,68 @@ resource "azurerm_sql_database" "sql-database" {
...
+    family   = "Gen4"
+  }
+
+  storage_profile {
+    storage_mb            = "5120"
+    backup_retention_days = "7"
+    geo_redundant_backup  = "Disabled"
+  }
+
+  administrator_login          = "${terraform.workspace}"
```
Comment (r-fennell) [inline terraform/azure.tf]: Please change this to simply `inspec-azure` - logins have max char of 16 and some workspaces will fail. Can you also please change it on line 625 in the `azurerm_sql_server` resource?

File: terraform/azure.tf [hunk 15]
```diff
@@ -635,6 +635,68 @@ resource "azurerm_sql_database" "sql-database" {
...
+  upper   = false
+}
+
+resource "azurerm_mysql_server" "mysql" {
+  name                = "${random_string.mysql_server.result}-mysqlsvr"
+  location            = "westeurope"
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+
+  sku {
+    name     = "B_Gen4_2"
```
Comment (r-fennell) [inline terraform/azure.tf]: Please change to Gen5

File: terraform/azure.tf [hunk 16]
```diff
@@ -635,6 +635,68 @@ resource "azurerm_sql_database" "sql-database" {
...
+    capacity = "2"
+    tier     = "Basic"
+    family   = "Gen4"
```
Comment (r-fennell) [inline terraform/azure.tf]: Please change to Gen5

**Comment by edwardbartholomew** [comment]
This PR has involved a lot of work - both code and reviewing - and is very worthwhile enhancement. Is there any chance of reviving?

**Comment by r-fennell** [comment]
I've added some of the functionality proposed in this PR in https://github.com/inspec/inspec-azure/pull/213. Hopefully the remaining work can be pushed over the line by the original author or @inspec/inspec-cloud-devs in the near future.

**Comment by mikaelkrief** [comment]
@r-fennell   I will review your comments

**Comment by rmoles** [comment]
Hi @mikaelkrief, thanks for the work on this PR, could you please resolve the conflicts?

**Comment by mikaelkrief** [comment]
@rmoles , Done conflicts resolved

File: docs/resources/azurerm_mysql_databases.md.erb [hunk 17]
```diff
@@ -72,6 +73,18 @@ name. This is a string value.
       it { should exist }
     end
 
+    
+* `location`
+
+### location
+
+Filters the results to include only those resource groups that reside ina given location. This is a string value.
```
Comment (rmoles) [inline docs/resources/azurerm_mysql_databases.md.erb]: `ina` --> `in a`

File: docs/resources/azurerm_network_interface.md.erb [hunk 18]
```diff
@@ -0,0 +1,127 @@
...
+    depends:
+      inspec-azure:
+        git: https://github.com/inspec/inspec-azure.git
+
+You'll also need to setup your Azure credentials; see the resource pack
+[README](https://github.com/inspec/inspec-azure#inspec-for-azure).
+
+### Version
+
+This resource first became available in 1.7.0 of the inspec-azure resource pack.
```
Comment (rmoles) [inline docs/resources/azurerm_network_interface.md.erb]: this should be targeted for 1.7.2 as the resource packed is currently at 1.7.1. Please update any instance of this is the other files.

Comment (mikaelkrief) [reply docs/resources/azurerm_network_interface.md.erb]: resolved

File: Rakefile [hunk 19]
```diff
@@ -18,6 +18,35 @@ REQUIRED_ENVS = %w{AZURE_CLIENT_ID AZURE_CLIENT_SECRET AZURE_TENANT_ID AZURE_SUB
 
 task default: :test
 desc 'Testing tasks'
+
+task test: %w{test:unit setup_env test:integration}
```
Comment (rmoles) [inline Rakefile]: It appears that a test task has recently been added to the rakefile. Would you be able to remove these changes?

File: libraries/azurerm_mysql_server.rb [hunk 20]
```diff
@@ -35,6 +35,7 @@ def initialize(resource_group: nil, server_name: nil)
   end
 
   def firewall_rules
+    management.mysql_server_firewall_rules(@resource_group, @server_name)
```
Comment (rmoles) [inline libraries/azurerm_mysql_server.rb]: This line should be removed. It has been replaced in a recent PR by the line below.

File: Rakefile (hunk 19 above)
Comment (mikaelkrief) [reply Rakefile]: removed

File: libraries/azurerm_mysql_server.rb (hunk 20 above)
Comment (mikaelkrief) [reply libraries/azurerm_mysql_server.rb]: fixed

**Comment by mikaelkrief** [comment]
@r-fennell  could you review this PR ?

File: terraform/azure.tf [hunk 21]
```diff
@@ -500,6 +500,68 @@ resource "azurerm_sql_database" "sql_database" {
...
+    geo_redundant_backup  = "Disabled"
+  }
+
+  administrator_login          = "inspec-azure"
+  administrator_login_password = "P4assw0rd!"
+  version                      = "5.7"
+  ssl_enforcement              = "Enabled"
+}
+
+resource "azurerm_mysql_database" "mysql" {
```
Comment (rmoles) [inline terraform/azure.tf]: Hi, @mikaelkrief I have just noticed that you have 2 `azurerm_mysql_server ` resources named the same `mysql`. In fact, the below `azurerm_mysql_firewall_rule` resource has also been names `mysql`. Could you update the naming of these resources please, as currently, the terraform will fail to run?

File: terraform/azure.tf [hunk 22]
```diff
@@ -485,10 +485,10 @@ resource "random_string" "sql" {
 
 resource "azurerm_sql_server" "sql_server" {
   name                         = "sql-srv-${random_string.sql.result}"
-  resource_group_name          = azurerm_resource_group.rg.name
-  location                     = var.location
-  version                      = var.sql-server-version
-  administrator_login          = "iazAdmin"
+  resource_group_name          = "${azurerm_resource_group.rg.name}"
```
Comment (rmoles) [inline terraform/azure.tf]: the quotes and brackets are not required. This style is deprecated as of terraform 0.12.

**Comment by mikaelkrief** [comment]
thanks @rmoles

**Pull request merged**